pkg/result/processors/sort_results.go

```diff
@@ -142,6 +200,94 @@ func (cmp ByColumn) Compare(a, b *result.Issue) compareResult {
 	return res
 }
 
+func (cmp *byColumn) String() string {
+	return comparatorToString("byColumn", cmp)
+}
+
+type byLinter struct{ next comparator }
+
+func (cmp *byLinter) Next() comparator { return cmp.next }
+
+func (cmp *byLinter) AddNext(c comparator) comparator {
+	cmp.next = c
+	return cmp
+}
+
+func (cmp *byLinter) Compare(a, b *result.Issue) compareResult {
+	var res compareResult
+
+	if res = compareResult(strings.Compare(a.FromLinter, b.FromLinter)); !res.isNeutral() {
+		return res
+	}
+
+	if next := cmp.Next(); next != nil {
+		return next.Compare(a, b)
+	}
+
+	return res
+}
+
+func (cmp *byLinter) String() string {
+	return comparatorToString("byLinter", cmp)
+}
+
+type bySeverity struct{ next comparator }
+
+func (cmp *bySeverity) Next() comparator { return cmp.next }
+
+func (cmp *bySeverity) AddNext(c comparator) comparator {
+	cmp.next = c
+	return cmp
+}
+
+func (cmp *bySeverity) Compare(a, b *result.Issue) compareResult {
+	var res compareResult
+
+	if res = severityCompare(a.Severity, b.Severity); !res.isNeutral() {
+		return res
+	}
+
+	if next := cmp.Next(); next != nil {
+		return next.Compare(a, b)
+	}
+
+	return res
+}
+
+func (cmp *bySeverity) String() string {
+	return comparatorToString("bySeverity", cmp)
+}
+
+func mergeComparator(cmps []comparator) (comparator, error) {
+	if len(cmps) == 0 {
+		return nil, errors.New("no comparator")
+	}
+
+	for i := 0; i < len(cmps)-1; i++ {
+		cmps[i].AddNext(cmps[i+1])
+	}
+
+	return cmps[0], nil
+}
+
+func severityCompare(a, b string) compareResult {
+	// The position inside the slice define the importance (lower to higher).
+	classic := []string{"low", "medium", "high", "warning", "error"}
+
+	if slices.Contains(classic, a) && slices.Contains(classic, b) {
```

Comment:
if some linters have "true" severities and some have custom severities, shouldn't we sort by groups – "true" first and then "custom" in lexigraphic order?

or maybe make doc stricter?
```yml
  # If the severity values are inside the following list, they are ordered in this order:
  # ...
  # Either they are sorted alphabetically.
```

```yml
  # If all severity values are inside the following list, they are ordered in this order:
```